app/models/supervised_experiment.rb

```diff
@@ -1,63 +1,115 @@
+##
+# This class represents an instance of supervised experiment.
+# SupervisedExperiment is a subclass of CustomPointsExperiment, with is subclass
+# of Experiment. The main difference from normal experiment is that in case of
+# supervised experiment scheduling of new simulation is external action performed
+# by supervisor script maintaining by experiment supervisor. This type of experiment
+# have also result, which is result of script that supervisors this experiment e.g.
+# some value or point from input space.
+#
+# List of possible attributes:
+# * supervised - boolean, always true, marker which allow to distinguish normal experiment from
+#   supervised one [set by #initialize]
+# * completed - boolean, marker which allow to check if experiment is completed
+#   (which equals that supervisor script finished and sent result) [set false by #initialize,
+#   modified to true by #mark_as_complete!]
+# * results - hash, contains result of experiment (sent by supervisor script) [set empty by
+#   initialize, modified by #mark_as_complete!]
+# * supervisor_script_uuid - string, id of supervisor script, used for authentication [set by
+#   #start_supervisor_script]
+# * is_error - boolean, this flag is true when experiment is in error state
+#   [set by ExperimentController#mark_as_complete]
+# * error_reason - string, set with is_error flag is_reason [set by ExperimentController#mark_as_complete]
 class SupervisedExperiment < CustomPointsExperiment
 
-  def init_empty(simulation)
-    super simulation
-    self.supervised = true
-    self.completed = false
-    self.result = {}
+  ##
+  # Sets parameters needed for SupervisorExperiment (described above), and calls super.
+  def initialize(attributes = {})
```

Comment:
"initialize" jest wywoływane za każdym razem kiedy rekord jest pobierany z bazy. Jeśli rekord nie ma pola "supervised" to je nadpiszesz na true. To pole powinno być ustawiane tylko w przypadku, kiedy tworzymy nowy rekord - dlatego powstała metoda "init_empty", która nie jest częścią konstruktora i którą się wywołuje ręcznie. Z kolei, żeby nie musieć jej wywoływać ręcznie powstała fabryka ExperimentFactory. 
Initialize można traktować jako deserializację: hash z bazy -> obiekt.

Tutaj ew. można rzucać wyjątek w przypadku (type != 'manual_points' lub supervised != true) - tylko trzeba by wtedy pobawić się z metodą create_supervised_experiment w fabryce (ustawianie params zamiast init_empty).

Jaką miałeś motywację, żeby zrezygnować z init_empty na rzecz initializera?